src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -705,17 +706,23 @@ public AssetSpecification buildAssetFromSnapshotAccessRequest(
     assetModel.addTablesItem(new AssetTableModel().name("person"));
 
     // Build asset tables and columns based on the concept sets included in the snapshot request
-    List<SnapshotBuilderTable> tables = pullTables(snapshotRequestModel);
+    List<SnapshotBuilderTable> tables = pullTables(snapshotAccessRequest);
     tables.forEach(
         table -> {
           assetModel.addTablesItem(
               new AssetTableModel().name(table.getDatasetTableName()).columns(table.getColumns()));
           // First add all person <-> occurrence relationships
-          assetModel.addFollowItem(table.getPrimaryTableRelationship());
+          if (table.getPrimaryTableRelationship() != null) {
+            assetModel.addFollowItem(table.getPrimaryTableRelationship());
+          }
         });
     // Second, add all occurrence <-> concept relationships
     tables.forEach(
-        table -> table.getSecondaryTableRelationships().forEach(assetModel::addFollowItem));
+        table -> {
+          if (table.getSecondaryTableRelationships() != null) {
+            table.getSecondaryTableRelationships().forEach(assetModel::addFollowItem);
```

Comment:
I added the null check here too because when we do include the sample table, it won't have secondary relationships which will cause the same issue I saw above for the person table.